src/coreclr/pal/src/thread/thread.cpp

```diff
@@ -1681,7 +1681,17 @@ CorUnix::InternalSetThreadDescription(
         nameBuf[15] = '\0';
```

Comment:
Does Apple has 15 char limit on the thread name, or is that limit Linux specific?

Reply:
It looks like the limit is 64 characters on MacOS - it's not publicly defined anywhere but apple publishes the headers where it's defined (https://opensource.apple.com/source/libpthread/libpthread-218.1.3/src/internal.h.auto.html). Do we want to support that as well? I think it should be pretty easy to implement.

Reply:
Yes, we should support. What does the Apple API does when the thread name is longer? Does it fail or does it truncate the name?

Reply:
It's going to truncate the name (https://github.com/apple/darwin-libpthread/blob/fe48dac35eb1232b0aa67260e79911387364568d/src/pthread.c#L1131)